### PR TITLE
PR: gh#62 : Adding support for multiple profiles

### PR DESCRIPTION
**Pull request by kanjoe24**
Adding support for multiple profiles

### CODE REVIEW

**Comment by kanjoe24** [comment]
Test Log:

```shell

***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: s

--------------------- Registered Suites -----------------------------
  #  Suite Name                                                       Init? Cleanup? #Tests Active?

  1. L1 - ut_control function tests                                    Yes      Yes      3     Yes
  2. L2 - ut_control Module tests                                      Yes      Yes      5     Yes
  3. L1 - ut_control mapping tests                                     Yes      Yes      2     Yes
  4. ut-kvp - test functions                                           Yes      Yes      2     Yes
  5. ut-kvp - test main functions YAML Decoder                         Yes      Yes     11     Yes
  6. ut-kvp - test main functions JSON Decoder                         Yes      Yes     10     Yes
  7. ut-kvp - test main functions Test without Open                    Yes      Yes      1     Yes
  8. ut-kvp - test main functions YAML Decoder with malloc'd data      Yes      Yes     10     Yes
  9. ut-kvp - test main functions JSON Decoder with malloc'd data      Yes      Yes      9     Yes
 10. ut-kvp - test kvp_open_memory()                                   Yes      Yes      1     Yes
 11. ut-kvp - test main functions YAML Decoder for includes using bu   Yes      Yes      3     Yes
 12. ut-kvp - test main functions YAML Decoder for single include fi   Yes      Yes      3     Yes
 13. ut-kvp - test main functions YAML Decoder for Yaml include supp   Yes      Yes      2     Yes
 14. ut-kvp - test main functions YAML Decoder for Yaml multiple pro   Yes      Yes      1     Yes
---------------------------------------------------------------------
Total Number of Suites : 14

Enter number of suite to select (1-14) : 14
Suite 'ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs' selected.

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: r

2024-12-10-16:30:54, LOG   , ut_console.c,   633 : 

2024-12-10-16:30:54, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs
2024-12-10-16:30:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp multiple profile'
2024-12-10-16:30:54, STEP  , ut_test_kvp.c,   750 : ut_kvp_open( pInstance, KVP_VALID_TEST_YAML_CONFIG_FILE )
2024-12-10-16:30:54, STEP  , ut_test_kvp.c,   754 : ut_kvp_open( pInstance, KVP_VALID_TEST_SINGLE_INCLUDE_FILE_YAML )
2024-12-10-16:30:54, STEP  , ut_test_kvp.c,   758 : ut_kvp_open( pInstance, KVP_VALID_TEST_DEPTH_CHECK_INCLUDE_YAML )---
decodeTest:
  checkUint8IsDeHex: 0xde
  checkUint8IsDeDec: 222
  checkUint16IsDeadHex: 0xdead
  checkUint16IsDeadDec: 57005
  checkUint32IsDeadBeefHex: 0xdeadbeef
  checkUint32IsDeadBeefDec: 3735928559
  checkUint64IsDeadBeefHex: 0xdeadbeefdeadbeef
  checkUint64IsDeadBeefDec: 16045690984833335023
  checkStringDeadBeef: "the beef is dead"
  checkStringDeadBeef2: the beef is also dead
  checkStringDeadBeefNoQuotes: the beef is dead
  checkBoolFalse: false
  checkBoolFALSE: FALSE
  checkBoolFaLse: FaLSE
  checkBoolTRUE: TRUE
  checkBooltrue: true
  checkBoolTRuE: TRuE
  checkStringList:
  - stringA
  - stringB
  - stringC
  checkUint32List:
  - 720
  - 800
  - 1080
  checkFloat: 5.1
  checkDoublePi: 3.14159265358979323846
  checkDoubleScientific: -4.2e8
  checkDoubleInvalid: invalid_string
  checkBytesSpace: 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01
  checkBytesComma: 0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b,0x2a, 0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b, 0x2a,0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85, 0x1b,0x2a,0x01,0x01,0x01,0x01
  checkBytesIncorrect: 00 ff ff ff ff ff ff 00 15 85 1b 2a 01 01 01 01 ff 20 01 03 80 7a 45 78 0a ee 91 a3 54 4c 99 26 0f 50 54 21 08 00 31 40 45 40 61 40 71 40 81 80 01 01 01 01 01 01 08 e8 00 30 f2 70 5a 80 b0 58 8a 00 40 84 63 00 00 1e 02 3a 80 18 71 38 2d 40 58 2c 45 00 40 84 63 00 00 1e 00 00 00 fd 00 17 3e 1e 88 3c 00 0a 20 20 20 20 20 20 00 00 00 fc 00 65 6c 65 6d 65 6e 74 20 54 56 0a 20 20 01 6f 02 03 5f f1 53 61 60 10 1f 04 13 05 14 03 02 12 20 21 22 15 01 5d 5e 5f 2c 09 57 03 15 07 50 57 07 00 67 7e 00 6d 03 0c 00 10 00 38 3c 20 00 60 01 02 03 68 d8 5d c4 01 78 88 0b 02 e2 00 cf e3 05 c0 00 e3 06 0d 01 e4 0f 03 00 00 eb 01 46 d0 00 4d 57 3a 94 34 07 a5 e6 11 46 d0 00 00 00 66 21 50 b0 51 00 1b 30 40 70 36 00 40 84 63 00 00 1e 00 00 00 00 00 00 00 00 00 00 00 00 00 00 66
  checkByteCommaSpaces: 0x00, 0xff, 0xff, 0xff
  checkByteSpaceSpaceCommaSpace: 0xff  0xdd,  0xee
  checkBytesDecimalSpace: 00 55 255 255
  checkBytesDecimalComma: 00,255,55,255
  checkBytesDecimalCommaSpace: 00, 255, 255, 55
  checkByteInvalid: 0xFF 9A
  checkBytePrefix: 0xFF 0xed 0X15 0xee
profile: DUMMY_PROFILE
hal version: X.X.X
components:
- name: ComponentManager1
  interfaceVersion: X
  ResourceList1:
  - id: 0
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
  - id: 1
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
- name: ComponentManager2
  interfaceVersion: X
  platformCapabilities:
  - supportsFeatureY: false
    sampleRateHz: 00000
    pcmFormat: DUMMY_FORMAT
    supportsAdditionalFormat: false
  ResourceList2:
  - id: 0
  - id: 1
- name: ComponentManager3
  interfaceVersion: X
  supportedModes:
  - DUMMY_MODE
  ResourceList3:
  - id: 0
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    - capability: DUMMY_CODEC_2
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    - RANGE_2
    - RANGE_3
    supportsFeatureZ: true
  - id: 1
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    supportsFeatureZ: true
"1":
  value: true
"2":
  value: true
"3":
  value: true
"4":
  value: true
"5":
  value: true


2024-12-10-16:30:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp multiple profile'

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites     14      1    n/a      0        0
               tests     63      1      1      0        0
             asserts      3      3      3      0      n/a

Elapsed time =    0.003 seconds

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: q
```

**Comment by kanjoe24** [comment]
Test script results:

```shell
./release-test-script-ut-control.sh -t feature/gh62-multiple-profile-input-support
==========================================================
RESULTS for ubuntu 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/linux/curl/lib/libcurl.a exists. PASS
build/linux/openssl/lib/libssl.a exists. PASS 
host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/arm/curl/lib/libcurl.a exists. PASS
build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/arm/curl/lib/libcurl.a exists. PASS
build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh62-multiple-profile-input-support. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================

```

File: src/ut_kvp.c [hunk 1]
```diff
@@ -116,7 +116,15 @@ ut_kvp_status_t ut_kvp_open(ut_kvp_instance_t *pInstance, char *fileName)
         UT_LOG_ERROR("[%s] cannot be accesed", fileName);
         return UT_KVP_STATUS_FILE_OPEN_ERROR;
     }
-    pInternal->fy_handle = fy_document_build_from_file(NULL, fileName);
+
+    if(pInternal->fy_handle)
```
Comment (Ulrond) [inline src/ut_kvp.c:120]: don't you also need to support this on `openMemory` function also?

File: tests/src/ut_test_kvp.c [hunk 2]
```diff
@@ -741,6 +743,34 @@ void test_ut_kvp_fieldPresent_on_main_yaml(void)
 
 }
 
+void test_ut_kvp_add_multiple_profile(void)
+{
+    ut_kvp_status_t status;
+
+    UT_LOG_STEP("ut_kvp_open( pInstance, KVP_VALID_TEST_YAML_CONFIG_FILE )");
+    status = ut_kvp_open( gpMainTestInstance, KVP_VALID_TEST_YAML_CONFIG_FILE);
```
Comment (Ulrond) [inline tests/src/ut_test_kvp.c:751]: need to extend testing to support `ut_kvp_openMemory` also

**Review by Ulrond — CHANGES_REQUESTED** [review]
probally should do this on `ut_kvp_openMemory` also

**Comment by kanjoe24** [comment]
logs from ut_control_test.sh


```shell
./ut_control_test.sh 
Listing Filename:[/tmp/ut-log_2024-12-17_155855.log-Listing.xml]
Results Filename:[/tmp/ut-log_2024-12-17_155855.log]

2024-12-17-15:58:55, LOG   , ut_cunit.c,   147 : ---- start of test run ----

 ut-core: Wrapper framework for testing frameworks

***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: s

--------------------- Registered Suites -----------------------------
  #  Suite Name                                                       Init? Cleanup? #Tests Active?

  1. L1 - ut_control function tests                                    Yes      Yes      3     Yes
  2. L2 - ut_control Module tests                                      Yes      Yes      5     Yes
  3. L1 - ut_control mapping tests                                     Yes      Yes      2     Yes
  4. ut-kvp - test functions                                           Yes      Yes      2     Yes
  5. ut-kvp - test main functions YAML Decoder                         Yes      Yes     11     Yes
  6. ut-kvp - test main functions JSON Decoder                         Yes      Yes     10     Yes
  7. ut-kvp - test main functions Test without Open                    Yes      Yes      1     Yes
  8. ut-kvp - test main functions YAML Decoder with malloc'd data      Yes      Yes     10     Yes
  9. ut-kvp - test main functions JSON Decoder with malloc'd data      Yes      Yes      9     Yes
 10. ut-kvp - test kvp_open_memory()                                   Yes      Yes      1     Yes
 11. ut-kvp - test main functions YAML Decoder for includes using bu   Yes      Yes      3     Yes
 12. ut-kvp - test main functions YAML Decoder for single include fi   Yes      Yes      3     Yes
 13. ut-kvp - test main functions YAML Decoder for Yaml include supp   Yes      Yes      2     Yes
 14. ut-kvp - test main functions YAML Decoder for Yaml multiple pro   Yes      Yes      2     Yes
---------------------------------------------------------------------
Total Number of Suites : 14

Enter number of suite to select (1-14) : 14
Suite 'ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs' selected.

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: s

----------------- Test List ------------------------------
Suite: ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs

 #  Test Name                          Active?

 1. kvp multiple profile                  Yes
 2. kvp multiple profile using open m     Yes
----------------------------------------------------------
Total Number of Tests : 2

Enter number of test to select (1-2) : 1

2024-12-17-15:59:08, LOG   , ut_console.c,   633 : 

2024-12-17-15:59:08, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs
2024-12-17-15:59:08, LOG   , ut_console.c,   639 :      Running Test : 'kvp multiple profile'
2024-12-17-15:59:08, STEP  , ut_test_kvp.c,   750 : ut_kvp_open( pInstance, KVP_VALID_TEST_YAML_CONFIG_FILE )
2024-12-17-15:59:08, STEP  , ut_test_kvp.c,   754 : ut_kvp_open( pInstance, KVP_VALID_TEST_SINGLE_INCLUDE_FILE_YAML )
2024-12-17-15:59:08, STEP  , ut_test_kvp.c,   758 : ut_kvp_open( pInstance, KVP_VALID_TEST_DEPTH_CHECK_INCLUDE_YAML )---
decodeTest:
  checkUint8IsDeHex: 0xde
  checkUint8IsDeDec: 222
  checkUint16IsDeadHex: 0xdead
  checkUint16IsDeadDec: 57005
  checkUint32IsDeadBeefHex: 0xdeadbeef
  checkUint32IsDeadBeefDec: 3735928559
  checkUint64IsDeadBeefHex: 0xdeadbeefdeadbeef
  checkUint64IsDeadBeefDec: 16045690984833335023
  checkStringDeadBeef: "the beef is dead"
  checkStringDeadBeef2: the beef is also dead
  checkStringDeadBeefNoQuotes: the beef is dead
  checkBoolFalse: false
  checkBoolFALSE: FALSE
  checkBoolFaLse: FaLSE
  checkBoolTRUE: TRUE
  checkBooltrue: true
  checkBoolTRuE: TRuE
  checkStringList:
  - stringA
  - stringB
  - stringC
  checkUint32List:
  - 720
  - 800
  - 1080
  checkFloat: 5.1
  checkDoublePi: 3.14159265358979323846
  checkDoubleScientific: -4.2e8
  checkDoubleInvalid: invalid_string
  checkBytesSpace: 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01
  checkBytesComma: 0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b,0x2a, 0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b, 0x2a,0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85, 0x1b,0x2a,0x01,0x01,0x01,0x01
  checkBytesIncorrect: 00 ff ff ff ff ff ff 00 15 85 1b 2a 01 01 01 01 ff 20 01 03 80 7a 45 78 0a ee 91 a3 54 4c 99 26 0f 50 54 21 08 00 31 40 45 40 61 40 71 40 81 80 01 01 01 01 01 01 08 e8 00 30 f2 70 5a 80 b0 58 8a 00 40 84 63 00 00 1e 02 3a 80 18 71 38 2d 40 58 2c 45 00 40 84 63 00 00 1e 00 00 00 fd 00 17 3e 1e 88 3c 00 0a 20 20 20 20 20 20 00 00 00 fc 00 65 6c 65 6d 65 6e 74 20 54 56 0a 20 20 01 6f 02 03 5f f1 53 61 60 10 1f 04 13 05 14 03 02 12 20 21 22 15 01 5d 5e 5f 2c 09 57 03 15 07 50 57 07 00 67 7e 00 6d 03 0c 00 10 00 38 3c 20 00 60 01 02 03 68 d8 5d c4 01 78 88 0b 02 e2 00 cf e3 05 c0 00 e3 06 0d 01 e4 0f 03 00 00 eb 01 46 d0 00 4d 57 3a 94 34 07 a5 e6 11 46 d0 00 00 00 66 21 50 b0 51 00 1b 30 40 70 36 00 40 84 63 00 00 1e 00 00 00 00 00 00 00 00 00 00 00 00 00 00 66
  checkByteCommaSpaces: 0x00, 0xff, 0xff, 0xff
  checkByteSpaceSpaceCommaSpace: 0xff  0xdd,  0xee
  checkBytesDecimalSpace: 00 55 255 255
  checkBytesDecimalComma: 00,255,55,255
  checkBytesDecimalCommaSpace: 00, 255, 255, 55
  checkByteInvalid: 0xFF 9A
  checkBytePrefix: 0xFF 0xed 0X15 0xee
profile: DUMMY_PROFILE
hal version: X.X.X
components:
- name: ComponentManager1
  interfaceVersion: X
  ResourceList1:
  - id: 0
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
  - id: 1
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
- name: ComponentManager2
  interfaceVersion: X
  platformCapabilities:
  - supportsFeatureY: false
    sampleRateHz: 00000
    pcmFormat: DUMMY_FORMAT
    supportsAdditionalFormat: false
  ResourceList2:
  - id: 0
  - id: 1
- name: ComponentManager3
  interfaceVersion: X
  supportedModes:
  - DUMMY_MODE
  ResourceList3:
  - id: 0
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    - capability: DUMMY_CODEC_2
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    - RANGE_2
    - RANGE_3
    supportsFeatureZ: true
  - id: 1
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    supportsFeatureZ: true
"1":
  value: true
"2":
  value: true
"3":
  value: true
"4":
  value: true
"5":
  value: true


2024-12-17-15:59:08, LOG   , ut_console.c,   661 :      Test Complete : 'kvp multiple profile'

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites     14      0    n/a      0        0
               tests     64      1      1      0        0
             asserts      3      3      3      0      n/a

Elapsed time =    0.003 seconds

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: 2

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: s

----------------- Test List ------------------------------
Suite: ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs

 #  Test Name                          Active?

 1. kvp multiple profile                  Yes
 2. kvp multiple profile using open m     Yes
----------------------------------------------------------
Total Number of Tests : 2

Enter number of test to select (1-2) : 2

2024-12-17-15:59:21, LOG   , ut_console.c,   633 : 

2024-12-17-15:59:21, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - test main functions YAML Decoder for Yaml multiple profile inputs
2024-12-17-15:59:21, LOG   , ut_console.c,   639 :      Running Test : 'kvp multiple profile using open memory'
2024-12-17-15:59:21, STEP  , ut_test_kvp.c,   778 : ut_kvp_openMemory( gpMainTestInstance, assets/config-test.yaml ) - Postive
2024-12-17-15:59:21, STEP  , ut_test_kvp.c,   785 : ut_kvp_openMemory( gpMainTestInstance, assets/include/single-include-file.yaml ) - Postive
2024-12-17-15:59:21, STEP  , ut_test_kvp.c,   792 : ut_kvp_openMemory( gpMainTestInstance, assets/include/depth_check.yaml ) - Postive---
decodeTest:
  checkUint8IsDeHex: 0xde
  checkUint8IsDeDec: 222
  checkUint16IsDeadHex: 0xdead
  checkUint16IsDeadDec: 57005
  checkUint32IsDeadBeefHex: 0xdeadbeef
  checkUint32IsDeadBeefDec: 3735928559
  checkUint64IsDeadBeefHex: 0xdeadbeefdeadbeef
  checkUint64IsDeadBeefDec: 16045690984833335023
  checkStringDeadBeef: "the beef is dead"
  checkStringDeadBeef2: the beef is also dead
  checkStringDeadBeefNoQuotes: the beef is dead
  checkBoolFalse: false
  checkBoolFALSE: FALSE
  checkBoolFaLse: FaLSE
  checkBoolTRUE: TRUE
  checkBooltrue: true
  checkBoolTRuE: TRuE
  checkStringList:
  - stringA
  - stringB
  - stringC
  checkUint32List:
  - 720
  - 800
  - 1080
  checkFloat: 5.1
  checkDoublePi: 3.14159265358979323846
  checkDoubleScientific: -4.2e8
  checkDoubleInvalid: invalid_string
  checkBytesSpace: 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01 0x00 0xff 0xff 0xff 0xff 0xff 0xff 0x00 0x15 0x85 0x1b 0x2a 0x01 0x01 0x01 0x01
  checkBytesComma: 0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b,0x2a, 0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85,0x1b, 0x2a,0x01,0x01,0x01,0x01,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0x00,0x15,0x85, 0x1b,0x2a,0x01,0x01,0x01,0x01
  checkBytesIncorrect: 00 ff ff ff ff ff ff 00 15 85 1b 2a 01 01 01 01 ff 20 01 03 80 7a 45 78 0a ee 91 a3 54 4c 99 26 0f 50 54 21 08 00 31 40 45 40 61 40 71 40 81 80 01 01 01 01 01 01 08 e8 00 30 f2 70 5a 80 b0 58 8a 00 40 84 63 00 00 1e 02 3a 80 18 71 38 2d 40 58 2c 45 00 40 84 63 00 00 1e 00 00 00 fd 00 17 3e 1e 88 3c 00 0a 20 20 20 20 20 20 00 00 00 fc 00 65 6c 65 6d 65 6e 74 20 54 56 0a 20 20 01 6f 02 03 5f f1 53 61 60 10 1f 04 13 05 14 03 02 12 20 21 22 15 01 5d 5e 5f 2c 09 57 03 15 07 50 57 07 00 67 7e 00 6d 03 0c 00 10 00 38 3c 20 00 60 01 02 03 68 d8 5d c4 01 78 88 0b 02 e2 00 cf e3 05 c0 00 e3 06 0d 01 e4 0f 03 00 00 eb 01 46 d0 00 4d 57 3a 94 34 07 a5 e6 11 46 d0 00 00 00 66 21 50 b0 51 00 1b 30 40 70 36 00 40 84 63 00 00 1e 00 00 00 00 00 00 00 00 00 00 00 00 00 00 66
  checkByteCommaSpaces: 0x00, 0xff, 0xff, 0xff
  checkByteSpaceSpaceCommaSpace: 0xff  0xdd,  0xee
  checkBytesDecimalSpace: 00 55 255 255
  checkBytesDecimalComma: 00,255,55,255
  checkBytesDecimalCommaSpace: 00, 255, 255, 55
  checkByteInvalid: 0xFF 9A
  checkBytePrefix: 0xFF 0xed 0X15 0xee
profile: DUMMY_PROFILE
hal version: X.X.X
components:
- name: ComponentManager1
  interfaceVersion: X
  ResourceList1:
  - id: 0
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
  - id: 1
    dummyCapabilities:
    - DUMMY_CAPABILITY_1
    - DUMMY_CAPABILITY_2
    - DUMMY_CAPABILITY_3
    - DUMMY_CAPABILITY_4
    supportsFeatureX: true
- name: ComponentManager2
  interfaceVersion: X
  platformCapabilities:
  - supportsFeatureY: false
    sampleRateHz: 00000
    pcmFormat: DUMMY_FORMAT
    supportsAdditionalFormat: false
  ResourceList2:
  - id: 0
  - id: 1
- name: ComponentManager3
  interfaceVersion: X
  supportedModes:
  - DUMMY_MODE
  ResourceList3:
  - id: 0
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    - capability: DUMMY_CODEC_2
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    - RANGE_2
    - RANGE_3
    supportsFeatureZ: true
  - id: 1
    dummyCapabilities:
    - capability: DUMMY_CODEC_1
      maxFrameRate: XX
      maxFrameWidth: XXXX
      maxFrameHeight: XXXX
    supportedRanges:
    - RANGE_1
    supportsFeatureZ: true
"1":
  value: true
"2":
  value: true
"3":
  value: true
"4":
  value: true
"5":
  value: true


2024-12-17-15:59:21, LOG   , ut_console.c,   661 :      Test Complete : 'kvp multiple profile using open memory'

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites     14      0    n/a      0        0
               tests     64      1      1      0        0
             asserts      3      3      3      0      n/a

Elapsed time =    0.003 seconds

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: q

2024-12-17-15:59:26, LOG   , ut_cunit.c,   177 : Logfile:[/tmp/ut-log_2024-12-17_155855.log]
2024-12-17-15:59:26, LOG   , ut_cunit.c,   185 : ---- end of test run ----

```

**Review by Ulrond — APPROVED** [review]
looks good

**Pull request merged**